### DISCUSSION TITLE
Sketchfab model won't resolve in Spoke

### DISCUSSION THREAD

**Discussion by hilarytsai** [mozilla/hubs#2791] [General Questions & Discussion] [answered]
Apologies for making so many posts lately, as well as the fact that this may or may not be more of a Sketchfab issue. I've reached out to their forum as well, in case it's nothing to do with Spoke.

I test uploaded a model (I have a free account) to Sketchfab for the purposes of importing it into my Mozilla Spoke room. I added tags, made it downloadable, and uploaded with no problems. In Spoke, however, when I pasted the newly uploaded model’s Url into the asset Url slot, my model would not resolve. I got a broken media image instead. I know it’s not to do with the model itself, as it looks right on sketchfab. However, Mozilla Spoke doesn’t recognize the Url for some reason. Is this a limitation of the free sketchfab account?  It’s my understanding that public, downloadable assets are a free feature.  In general, I've also noticed that many Sketchfab assets (not just mine) do not resolve in Spoke.  What usually causes that?

**Comment by emclaren** [comment]
Hi there, the first thing I always check when troubleshooting Sketchfab model issues with Hubs/Spoke is whether the object is marked as downloadable - if it's not downloadable our servers aren't able to access the model information. You should be able to upload downloadable assets as a free feature in Sketchfab. By the sounds of it, it seems like you've shared the file properly on Sketchfab. 

Th next things that come to mind are that perhaps there is an issue with the file's size, or the file format (sometimes there are issues with Tiltbrush files uploaded to Google Poly - though this isn't an issue I've seen with Sketchfab). 

Is there any chance you could share an example of a model that isn't working for you?

**Reply by hilarytsai** [reply]
Thank you so much, @emclaren!  I can confirm the model is downloadable. The asset is very small (it's a simple cube), with just a color channel and no image texture.  The size is only 3KB.  I just noticed something interesting, though.  When I search for it through Sketchfab in Spoke, I can drag my model into the scene. It's only when I paste the Url into the Url slot of the model element (or spawner) that it doesn't work.   Do you know why this could be? I'll link the Url to my test upload: https://sketchfab.com/3d-models/mysketchfabuploadtest-4e809280d9514c74bcfe330b54e5d97b

edit:  I tried another upload, this time my format was glTF (not glb-binary like my last upload) Since Sketchfab free account does not allow auto glTF conversion, I went ahead and exported as glTF.  The funny thing is that the glTF version is the only one that shows up in Spoke's Sketchfab searchbar.  The glb-binary one does not show up.  So to sum up, the glTF version can be dragged into the scene from Sketchfab search, and it works; but when I take the Url from Sketchfab and paste it into spawner for example, it does not import correctly.

**Comment by j-conrad** [comment, marked as answer]
I just tried the model you provided. It seemed to work fine--although it's really really large.
The model comes in with its scale set to 0.1, 0.1, 0.1 and each side of the cube is about 2000m.
Is it possible you just didn't see it due to its size?
![image](https://user-images.githubusercontent.com/31968496/89353314-fbf16480-d66a-11ea-9dc6-cd67957c43e9.png)

**Reply by hilarytsai** [reply]
Hi, @j-conrad and thank you!  Wow.. you're right about the size.  I used Cinema 4D's default settings (just created cube and didn't scale down or anything), so that's something I need to pay attention to. 

 I am able to drag in my asset from the Sketchfab search bar in Spoke and have it show up successfully; but pasting the url into the spawner or model elements just weren't working.   However, something tells me my issue is not with the asset, but something else. 
 I noticed just now that I get a broken media image every time I import the model element or spawner (with just the default duck, not even with pasting in my asset's Url)--so it actually may not be a problem with the test asset after all.  I'm stumped on why the regular duck model doesn't show up though.

<img width="1487" alt="error loading" src="https://user-images.githubusercontent.com/40791828/89356781-732af680-d673-11ea-9a07-0d6ceb101b72.png">

**Reply by j-conrad** [reply]
That is strange. The way I brought your model in was by first placing the generic (duck) model element, then replacing the URL with yours.
I'm wondering if maybe there's some limit you're hitting on Sketchfab's servers.
Either way, I'm glad you were able to get your model in one way or another.

**Reply by hilarytsai** [reply]
Thank you for all your help, @j-conrad and @emclaren !  Yes, I agree it's strange.  I don't think I reached any limit because I haven't done much on that account.  However, I will try again after restarting Chrome.  Sometimes that resets things.

**Comment by j-conrad** [comment]
It's also worth noting that it takes some time for a model to get auto-converted to glTF on Sketchfab. I've seen it convert anywhere from 2 minutes to over an hour. I'm not sure what makes it faster or slower.

**Reply by hilarytsai** [reply]
Thanks for this note!!  After I read this, I checked again and saw that my other glb's had shown up in the Sketchfab search, so I do think it was just a delay

**Reply by soorajkbabu** [reply]
@hilarytsai the strange issue still exists 🗡️ 
Were you able to solve the issue? 
I am using spoke locally and I assume it is a CORS issue.
